### PR TITLE
docs: extend docker image documentation by unreleased splunk

### PR DESCRIPTION
**Pull request by mkolasinski-splunk**
Add documentation regarding docker images used in reusable workflow with unreleased splunk.

### CODE REVIEW

**Review by artemrys — APPROVED** [review]

File: runbooks/addonfactory-workflow-addon-release-docker-images.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+- trigger [pipeline](https://cd.splunkdev.com/taautomation/ta-automation-docker-images/-/pipelines/new) for every OS flavor separately
+
+## Runbook to publish unreleased splunk image for testing
+Whenever there is a need for running tests with unreleased splunk, we need to create relevant splunk docker image and publish it to aws ecr
+### Steps
+#### Build docker image and publish to artifactory
+- Prior creating docker image it needs to be determined which revision of core splunk repo is required. Splunk docker images are based on splunk builds published to artifactory by CI in core repository. Their names match SHA of the commit in core repo: [develop builds artifactory](https://repo.splunkdev.net:443/artifactory/generic/splcore/builds/develop/)
+- Docker image is built by [pipeline](https://cd.splunkdev.com/core-ee/docker-splunk-internal/-/pipelines/new) which required UNRELEASED_SPLUNK_SHA as an input variable - provide first 12 characters of desired revision on splunk core repo. Once image is built, it is published to [artifactory](https://repo.splunkdev.net/ui/repos/tree/General/docker/docker-splunk-internal/unreleased/splunk-redhat-9).
```
Comment (kdoroszko-splunk) [inline runbooks/addonfactory-workflow-addon-release-docker-images.md:19]: The link does not work.

Comment (mkolasinski-splunk) [reply runbooks/addonfactory-workflow-addon-release-docker-images.md:19]: @kdoroszko-splunk did you log in to artifactory?

File: runbooks/addonfactory-workflow-addon-release-docker-images.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Runbook to creating and publishing docker images used in reusable workflow
+## Runbook to publish multiple images of different Linux flavors and versions for scripted inputs tests
+Once there is new Splunk release, and [matrix](https://github.com/splunk/addonfactory-test-matrix-action) is updated, we need to make sure that Splunk images for scripted inputs tests are created and published.
+### Steps
+
+#### Update OS images
+- check what OS are listed in definition of matrix in scripted inputs tests [here](https://github.com/splunk/addonfactory-workflow-addon-release/blob/v4.16/.github/workflows/reusable-build-test-release.yml#L1966)
```
Comment (kdoroszko-splunk) [inline runbooks/addonfactory-workflow-addon-release-docker-images.md:7]: Does the link point to the proper line?

File: runbooks/addonfactory-workflow-addon-release-docker-images.md (hunk 1 above)
Comment (kdoroszko-splunk) [reply runbooks/addonfactory-workflow-addon-release-docker-images.md:19]: Yes, artifactory links work, 
This one is faulty: https://cd.splunkdev.com/core-ee/docker-splunk-internal/-/pipelines/new

Comment (kdoroszko-splunk) [reply runbooks/addonfactory-workflow-addon-release-docker-images.md:19]: Yes, I have access now.

File: runbooks/addonfactory-workflow-addon-release-docker-images.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+## Runbook to publish unreleased splunk image for testing
+Whenever there is a need for running tests with unreleased splunk, we need to create relevant splunk docker image and publish it to aws ecr
+### Steps
+#### Build docker image and publish to artifactory
+- Prior creating docker image it needs to be determined which revision of core splunk repo is required. Splunk docker images are based on splunk builds published to artifactory by CI in core repository. Their names match SHA of the commit in core repo: [develop builds artifactory](https://repo.splunkdev.net:443/artifactory/generic/splcore/builds/develop/)
+- Docker image is built by [pipeline](https://cd.splunkdev.com/core-ee/docker-splunk-internal/-/pipelines/new) which required UNRELEASED_SPLUNK_SHA as an input variable - provide first 12 characters of desired revision on splunk core repo. Once image is built, it is published to [artifactory](https://repo.splunkdev.net/ui/repos/tree/General/docker/docker-splunk-internal/unreleased/splunk-redhat-9).
+#### Pull built image locally, tag and publish to ecr
+- docker pull docker.repo.splunkdev.net/docker-splunk-internal/unreleased/splunk-redhat-9:[image-tag]
```
[the remark this answers is not in the document]
Comment (mkolasinski-splunk) [reply runbooks/addonfactory-workflow-addon-release-docker-images.md:21]: Sure

File: runbooks/addonfactory-workflow-addon-release-docker-images.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+#### Update OS images
+- check what OSs are listed in definition of matrix for scripted inputs tests [here](https://github.com/splunk/addonfactory-workflow-addon-release/blob/72497e5c03894369b8fbdd2a2c4134c233ba1b5d/.github/workflows/reusable-build-test-release.yml#L27)
+- if any is missing in [ta-automation-docker-images](https://cd.splunkdev.com/taautomation/ta-automation-docker-images/-/tree/main/dockerfiles) then add new Dockerfile
+
+#### Create images and publish them to ECR
+- figure out what version of Splunk is needed (sha) using go/fetcher
+- trigger [pipeline](https://cd.splunkdev.com/taautomation/ta-automation-docker-images/-/pipelines/new) for every OS flavor separately
+
+## Runbook to publish unreleased splunk image for testing
```
Comment (kdoroszko-splunk) [inline runbooks/addonfactory-workflow-addon-release-docker-images.md:14]: `Splunk` instead of `splunk`? Same in the following lines?

Comment (mkolasinski-splunk) [reply runbooks/addonfactory-workflow-addon-release-docker-images.md]: Done.

**Review by kdoroszko-splunk — APPROVED** [review]

**Pull request merged**